bin/serve.dart

```diff
@@ -18,25 +18,17 @@ Future<void> main() async {
   print('Serving at http://${server.address.host}:${server.port}');
 }
 
+// By default allow access from everywhere.
+const _corsHeaders = <String, String>{'Access-Control-Allow-Origin': '*'};
+
 /// Middleware which adds [CORS headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS)
 /// to shelf responses. Also handles preflight (OPTIONS) requests.
-Middleware createCorsHeadersMiddleware({Map<String, String> corsHeaders}) {
-  // By default allow access from everywhere.
-  corsHeaders ??= <String, String>{'Access-Control-Allow-Origin': '*'};
-
-  // Handle preflight (OPTIONS) requests by just adding headers and an empty
-  // response.
-  Response handleOptionsRequest(Request request) {
-    if (request.method == 'OPTIONS') {
-      return Response.ok(null, headers: corsHeaders);
-    } else {
-      return null;
-    }
-  }
-
-  Response addCorsHeaders(Response response) =>
-      response.change(headers: corsHeaders);
-
-  return createMiddleware(
-      requestHandler: handleOptionsRequest, responseHandler: addCorsHeaders);
-}
+Handler _corsHeadersMiddleware(Handler innerHandler) => (request) async {
+      if (request.method == 'OPTIONS') {
+        return Response.ok(null, headers: _corsHeaders);
+      }
+
+      final response = await innerHandler(request);
+
+      return response.change(headers: _corsHeaders);
```

Comment:
To be more careful, etc – we could check if the response already has `Access-Control-Allow-Origin` header and only set it if it's not there.